alerter.c

```diff
@@ -1,31 +1,46 @@
 #include <stdio.h>
 #include <assert.h>
 
+//Use this macro to distiguish code for production and test environment
+#define TEST_ENVIRONMENT 1

```

Comment:
Using conditional compilation for separating production and test code may get complicated as the code size grows.
it is better to use interfaces or functions pointers or lambda functions (if available) to inject either required production or test code based on the envt.
Production code should be simple in the sense, given an input the output should be deterministic, may it be dev environment or test or field